src/main/java/sparksoniq/jsoniq/compiler/translator/expr/flowr/LetClause.java

```diff
@@ -55,6 +56,15 @@ public void setPreviousClause(FlworClause previousClause) {
         letVars.get(0).previousClause = this.previousClause;
     }
 
+    @Override
+    public void initHighestExecutionMode() {
+        // call isDataFrame on the last letVariable
+        this._highestExecutionMode =
+            letVars.get(letVars.size() - 1).getHighestExecutionMode().isDataFrame()
+                ? ExecutionMode.DATAFRAME
```

Comment:
The same applies here: why not

this._highestExecutionMode = letVars.get(letVars.size() - 1).getHighestExecutionMode()

?

Reply:
Also fixed here